kolibri/utils/server.py

```diff
@@ -322,13 +318,12 @@ def run(self):
         # If set of addresses that were present at the last time zeroconf updated its broadcast list
         # don't match the current set of all addresses for this device, then we should reinitialize
         # zeroconf, the listener, and the broadcast kolibri service.
-        current_addresses = get_all_addresses()
-        if self.addresses != current_addresses:
+        current_addresses = set(get_all_addresses())
+        if self.broadcast is not None and self.broadcast.addresses != current_addresses:
```

Comment:
Should we check `self.broadcast.is_broadcasting` here as well? As otherwise `self.broadcast.addresses` will always be an empty set.

Reply:
Good call